frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/logic/marketingAnalyticsLogic.ts

```diff
@@ -37,6 +37,56 @@ import {
     validColumnsForTiles,
 } from './utils'
 
+function getSourceStatus(
+    source: { id: string; name: string; type: string; prefix?: string },
+    nativeSources: ExternalDataSource[],
+    validExternalTables: ExternalTable[]
+): { status: 'Completed' | 'Failed' | 'Running' | 'warning' | 'error'; message: string } {
```

Comment:
We could use a `StatusSource` type here, but really nit, I just also tend to create those and the autocomplete works but when we're reviewing the magic strings look weird. It is the same with insight props, but I think we should default to use more standard data structure/types